lib/bgpstream_elem.h

```diff
@@ -27,6 +27,10 @@
 #ifndef __BGPSTREAM_ELEM_H
 #define __BGPSTREAM_ELEM_H
 
+#include "config.h"
```

Comment:
Without the include, the compiler can't look up the WITH_RPKI variable from configure.ac

Reply:
Unfortunately @digizeph is correct. Any header that is installed (i.e., contains part of the "public" API) **cannot** include `config.h` since it is only available at compilation time.

We have two options: 
1. we can script the building of the header file based on output from `configure`, but the problem with this is that we're talking about changing the fields in a public structure based on compile-time options, so this will make life hard for users.
1. we always include all the possible fields in the public structures. this is much better since the API will be stable no matter what compile time options are used.

Reply:
Alright! I didn't know how to solve this problem in an elegant way because the ```bgpstream_annotations``` struct contained a pointer to a struct of the shared library which is only available if the WITH_RPKI option is set. I have now used a void pointer instead. As a result there is no include of the shared library header in ```bgpstream_elem.h``` and no ifdef statement any more. In addition, ```bgpstream_utils_rpki.c``` and ```bgpstream_utils_rpki.h``` are only compiled if the option is set to avoid the ifdef-statements. 

Is this solution suitable for you?

Reply:
Yeah, that is fine, though it might be a little easier if you use an opaque struct pointer. That way you only need the struct definition when you try to dereference the pointer.

Something like:
```
struct struct_rpki_config_t *cfg;
```
might work. Maybe.

Reply:
Works like a charm. Thanks.